library/core/src/main/java/com/google/android/exoplayer2/extractor/ts/TsExtractor.java

```diff
@@ -498,6 +500,10 @@ private EsInfo readEsInfo(ParsableByteArray data, int length) {
         } else if (descriptorTag == TS_PMT_DESC_ISO639_LANG) {
           language = new String(data.data, data.getPosition(), 3).trim();
           // Audio type is ignored.
+        } else if (descriptorTag == TS_PMT_DESC_DVBSUBS) {
+          streamType = TS_STREAM_TYPE_DVBSUBS;
+          // we only support one subtitle service per PID
```

Comment:
It's there to signal that we are only processing part of the descriptor, the subtitling decriptor is defined as:
```
subtitling_descriptor(){
  descriptor_tag                8 uimsbf
  descriptor_length             8 uimsbf
  for (i= 0;i<N;i++){
    ISO_639_language_code      24 bslbf
    subtitling_type             8 bslbf
    composition_page_id        16 bslbf
    ancillary_page_id          16 bslbf
  }
}
```

In our services and all the examples I had found they used only one subtitle per PID, but in the generic case the parsing is incorrect and we only process the first subtitle for this PID.